pip_audit/_dependency_source/requirement.py

```diff
@@ -313,12 +318,21 @@ def _collect_preresolved_deps(
             if req.marker is not None and not req.marker.evaluate():
                 continue  # pragma: no cover
 
-            # This means we have a duplicate requirement for the same package
-            if req.name in req_names:
+            duplicate_req_specifier = req_specifiers.get(req.name)
+
+            if not duplicate_req_specifier:
+                req_specifiers[req.name] = req.specifier
+
+            # We have a duplicate requirement for the same package
+            # but different specifiers, meaning a badly resolved requirements.txt
+            elif duplicate_req_specifier != req.specifier:
                 raise RequirementSourceError(
                     f"package {req.name} has duplicate requirements: {str(req)}"
                 )
-            req_names.add(req.name)
+            else:
+                # We have a duplicate requirement for the same package and the specifier matches
+                # As they would return the same result from the audit, there no need to yield it a second time.
+                continue
```

Comment:
From CI: this line needs coverage!

Reply:
This is weird, coverage complains about this line only with py3.8 and 3.9, from 3.10 and above I get 100%.

Reply:
This is probably a manifestation of https://github.com/nedbat/coveragepy/issues/198, which was ultimately resolved in 3.10+ via PEP 626: https://peps.python.org/pep-0626/

The two solutions here are either to drop the `else: ... continue` *or* add a coverage ignore comment.

Reply:
Unfortunately, dropping the `else` cant be done without also changing the surrounding code, it's part of the logic.

I add an ignore comment, hope it will suffice 🤞